cobc/parser.y

```diff
@@ -13845,6 +13862,54 @@ inspect_format_variant:
 | inspect_converting
 ;
 
+/* EXAMINE statement */
+
+examine_statement:
+  EXAMINE
+  {
+	begin_statement ("EXAMINE", 0);
+  }
+  send_identifier
+  examine_format_variant
+;
+
+examine_format_variant:
+  TALLYING
+  {
+	cb_init_tallying ();
+	cb_build_tallying_data (cb_build_identifier (cb_build_reference ("TALLY"), 0));
+  }
+  examine_tallying_keyword simple_display_value
+  _examine_tallying_replacing
+  {
+	cb_emit_examine_tallying ($0, $4, examine_keyword.tallying, $5);
+  }
+| REPLACING examine_replacing_keyword
+  simple_display_value BY simple_display_value
```

Comment:
`simple_display_value` seems wrong, I suggest to create a new entry in the kind of

```yacc
single_alphabet_lits:
  alphabet_lits
  {
	if (CB_LITERAL_P($1)) {
		if (CB_LITERAL($1)->length != 1) ...
	}
  }
```

This way you don't get a bunch of potential bad entries in the codegen (and also have a guaranteed literal or a constant in there later).

Reply:
GCOS allows identifiers in addition to single-character literals, so I've expanded a bit on your suggestion.

Reply:
looks mostly good; as ref-mod is explicit allowed you may want to check for ref-mod size, too (or disallow it)

Reply:
Ref-mod with given length of `1` given as a literal is handled (but is it in bytes). Not sure handling more cases, that may require adding checks at runtime, is absolutely necessary as `EXAMINE` may be kind of obsolete.

Reply:
Actually it's forbidden on GCOS, so I disallowed that case.